### PR TITLE
Support building using the manylinux docker image.

### PR DESCRIPTION
**Pull request by lpetre**
Two changes:
- libpython is only required for embedded python, not for python extensions.
- tell cmake explicitly to use the current interpreter (and includes)

See related issues:
- https://gitlab.kitware.com/cmake/cmake/-/issues/20425
- https://github.com/pypa/manylinux/issues/484

### CODE REVIEW

File: transformer_engine/CMakeLists.txt [hunk 1]
```diff
@@ -22,7 +22,12 @@ endif()
 list(PREPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake/")
 find_package(CUDAToolkit REQUIRED cublas nvToolsExt)
 find_package(CUDNN REQUIRED cudnn)
-find_package(Python COMPONENTS Interpreter Development REQUIRED)
+
+if(${CMAKE_VERSION} VERSION_LESS "3.18.0")
+  find_package(Python COMPONENTS Interpreter Development REQUIRED)
+else()
+  find_package(Python COMPONENTS Interpreter Development.Module REQUIRED)
+endif()
```
Comment (timmoon10) [inline transformer_engine/CMakeLists.txt:30]: We already require CMake >3.18:
```suggestion
find_package(Python COMPONENTS Interpreter Development.Module REQUIRED)
```

Comment (lpetre) [reply transformer_engine/CMakeLists.txt]: Great, thanks for pointing that out

**Review by timmoon10 — APPROVED** [review]
LGTM, thanks for the fix!

**Comment by timmoon10** [comment]
The errors are unrelated and they also showed up in the 1/12 nightly build.

**Pull request merged**